libs/deps/commerce.d.ts

```diff
@@ -0,0 +1,316 @@
+      type Filter = RequiredKey<Plugin, 'filter'>;
+      type Level = 'debug' | 'error' | 'info' | 'warn';
+
+      interface Entry {
+        level: Level;
+        message: string;
+        namespace: string;
+        params: any[];
+        source: string;
+        timestamp: number;
+      }
+
+      interface Instance {
+        readonly id: string;
+        readonly namespace: string;
+        debug(message: string, ...params: any[]): void;
+        error(message: string, ...params: any[]): void;
+        /**
+         * Creates and returns new instance of `Log`
+         * bound to source module specified by `name`.
+         * 
+         * New log instance will append `name` to namespace of its records.
+         */
+        module(name: string): Instance;
+        info(message: string, ...params: any[]): void;
+        warn(message: string, ...params: any[]): void;
+      }
+
+      interface Plugin {
+        append?(record: Entry): void;
+        filter?(record: Entry): boolean;
+      }
+
+      interface Root {
+        commerce: Instance;
+        level: Record<Level, string>;
+        milo: Instance;
+        /**
+         * Writes log records to devtools console.
+         * Used by default in `local` env.
+         */
+        consoleAppender: Plugin;
+        /**
+         * Filters out all debug records.
+         * Used by default in `prod` env.
+         */
+        debugFilter: Plugin;
+        /**
+         * Filters out all records.
+         * Useful in tests.
+         */
+        quietFilter: Plugin;
+        /**
+         * Sends log error records to Lana.
+         * Used by default in `prod` env.
+         */
+        lanaAppender: Plugin;
+        /**
+         * Registers built-in log plugins suitable for provided `env`.
+         * @param env Milo `config.env` object.
+         */
+        init(env?: { name: string }): void;
+        /**
+         * Resets all log plugins registrations.
+         */
+        reset(): void;
+        /**
+         * Registers given log plugins.
+         */
+        use(...plugins: Plugin[]): void;
+      }
+    }
+
+    module Wcs {
+      type PlanType = 'ABM' | 'PUF' | 'M2M' | 'PERPETUAL';
+      type Env = Environment;
+
+      interface Client {
+        /**
+         * Resolves requested list of "Offer Selector Ids" (`osis`) from Wcs or local cache.
+         * Returns one promise per osi, the promise resolves to array of product offers
+         * associated with this osi.
+         * 
+         * If `multiple` is set to false (this is default value), resolved array will contain only one
+         * offer, selected by country/language-perpetual algorithm.
+         * Otherwise. all responded offers are returned.
+         * 
+         * If `taxExclusive` is set to true (default value defined in settings),
+         * then returned prices are transformed into tax exclusive variant.
+         */
+        resolveOfferSelectors(options: {
```

Comment:
as discussed offline, i really love that way of documenting APIs without bundling any extra byte to the outside payload. Once this is merged, let's discuss to make that more spread ❤️